patches/lua-tree.patch

```diff
@@ -384,6 +385,37 @@ index c1a150a..3d6c639 100644
  local constants = require "kong.constants"
  
  
+@@ -109,18 +109,18 @@ end
+
+ local function validate_tag(tag)
+
+-  local ok, err = validate_utf8_string(tag)
+-  if not ok then
+-    return nil, err
+-  end
+-
+-  -- printable ASCII (33-126 except ','(44) and '/'(47),
+-  -- plus non-ASCII utf8 (128-244)
+-  if not match(tag, "^[\033-\043\045\046\048-\126\128-\244]+$") then
+-    return nil,
+-    "invalid tag '" .. tag ..
+-      "': expected printable ascii (except `,` and `/`) or valid utf-8 sequences"
+-  end
++  -- local ok, err = validate_utf8_string(tag)
++  -- if not ok then
++  --   return nil, err
++  -- end
++
++  -- -- printable ASCII (33-126 except ','(44) and '/'(47),
++  -- -- plus non-ASCII utf8 (128-244)
++  -- if not match(tag, "^[\033-\043\045\046\048-\126\128-\244]+$") then
++  --   return nil,
++  --   "invalid tag '" .. tag ..
++  --     "': expected printable ascii (except `,` and `/`) or valid utf-8 sequences"
++  -- end
+
+   return true
+ end
```

Comment:
This function is considered part of the public interface that plugins can consume and hence I'm a bit hesitant to accept it.
If strings. the match is the root cause, is using the [go.re2 module](https://cs.github.com/Kong/goks?q=go.re2) an option? That will still result in patch to this file, which is okay but it may be able to solve the problem for a wider base.

What do you think?

Reply:
Makes sense to me, I will experiment a bit on this and run some tests to make sure this is a viable option.

Reply:
Fixed up the code, but needs further testing; converting PR to draft for further evaluation.